GCC 12 has a false positive warning when compiled for some architectures, e.g. Intel Sapphire Rapids.
See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=112370

Suppress this warning such that the build doesn't error.
Also disable a false positive that produces a lot of warnings/output.

Author: Alexander Grund (TU Dresden)
diff --git a/CMakeLists.txt b/CMakeLists.txt
index b74bf4536f4..bb062fa843a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -653,6 +653,8 @@ if(MSVC)
   string(APPEND CMAKE_CXX_FLAGS " /FS")
   string(APPEND CMAKE_CUDA_FLAGS " -Xcompiler /FS")
 endif(MSVC)
+append_cxx_flag_if_supported("-Wno-free-nonheap-object" CMAKE_CXX_FLAGS)
+append_cxx_flag_if_supported("-Wno-dangling-reference" CMAKE_CXX_FLAGS)
 
 string(APPEND CMAKE_CUDA_FLAGS " -Xfatbin -compress-all")
 
